### PR TITLE
Add Unit test TestRenderTemplate to test operator/pkg/util/util.go RenderTempla…

### PR DESCRIPTION
**Pull request by kanapuli**
Added unit test to test the RenderTemplate() function in operator/pkg/util/util.go. I am new to istio contributions. Let me know if I did something wrong.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: operator/pkg/util/util_test.go [hunk 1]
```diff
@@ -182,3 +183,67 @@ func sameStringSlice(x, y []string) bool {
...
+			},
+			want: "",
+			err:  errors.New(""),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			got, err := RenderTemplate(tt.template, tt.in)
+
+			if want, wantErr := tt.want, tt.err; !(got == want) || ((err == nil && wantErr != nil) || (err != nil && wantErr == nil)) {
```
Comment (howardjohn) [inline operator/pkg/util/util_test.go:244]: This would be a bit more understanable if you split out the err checking from the expected output checking

**Review by howardjohn — COMMENTED** [review]
/ok-to-test

File: operator/pkg/util/util_test.go (hunk 1 above)
Comment (kanapuli) [reply operator/pkg/util/util_test.go]: @howardjohn The code is refactored to make it more understandable.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**